neurodsp/filt/utils.py

```diff
@@ -241,3 +256,96 @@ def remove_filter_edges(sig, filt_len):
     sig[-n_rmv:] = np.nan
 
     return sig
+
+
+def gen_filt_report(pass_type, filt_type, fs, f_db, db, pass_bw,
+                    transition_bw, f_range, f_range_trans):
+    """Create a filter report.
+
+    Parameters
+    ----------
+    pass_type : {'bandpass', 'bandstop', 'lowpass', 'highpass'}
+        Which type of filter was applied.
+    filt_type : str, {'FIR', 'IIR'}
+        The type of filter being applied.
+    fs : float
+        Sampling rate, in Hz.
+    f_db : 1d array
+        Frequency vector corresponding to attenuation decibels, in Hz.
+    db : 1d array
+        Degree of attenuation for each frequency specified in `f_db`, in dB.
+    pass_bw : float
+        The pass bandwidth of the filter.
+    transition_band : float
+        The transition bandwidth of the filter.
+    f_range : tuple of (float, float) or float
+        Cutoff frequency(ies) used for filter, specified as f_lo & f_hi.
+    f_range_trans : tuple of (float, float)
+        The lower and upper frequencies of the transition band.
+
+    Returns
+    -------
+    filt_report : dict
+        A dicionary of filter parameter keys and corresponding values.
+    """
+    filt_report = {}
+
+    # Filter type (high-pass, low-pass, band-pass, band-stop, FIR, IIR)
+    filt_report['Pass Type'] = '{pass_type}'.format(pass_type=pass_type)
+
+    # Cutoff frequenc(ies) (including definition)
+    filt_report['Cutoff (Half-Amplitude)'] = '{cutoff} Hz'.format(cutoff=f_range)
+
+    # Filter order (or length)
+    filt_report['Filter Order'] = '{order}'.format(order=len(f_db)-1)
+
+    # Roll-off or transition bandwidth
+    filt_report['Transition Bandwidth'] = '{:.1f} Hz'.format(transition_bw)
+    filt_report['Pass/Stop Bandwidth'] = '{:.1f} Hz'.format(pass_bw)
+
+    # Passband ripple and stopband attenuation
+    pb_ripple = np.max(db[:np.where(f_db < f_range_trans[0])[0][-1]])
+    sb_atten = np.max(db[np.where(f_db > f_range_trans[1])[0][0]:])
+    filt_report['Passband Ripple'] = '{pb_ripple} db'.format(pb_ripple=pb_ripple)
+    filt_report['Stopband Attenuation'] = '{sb_atten} db'.format(sb_atten=sb_atten)
+
+    # Filter delay (zero-phase, linear-phase, non-linear phase)
+    filt_report['Filter Type'] = filt_type
+
+    if filt_type == 'FIR':
+
+        filt_report['Phase'] = '{filt_class}'.format(filt_class='linear-phase')
+        filt_report['Group Delay'] = '{delay}s'.format(delay=(len(f_db)-1) / (2 * fs))
+        filt_report['Direction'] = 'one-pass reverse'
```

Comment:
why reverse? I'm not sure about that

Reply:
I was confused about the terminology here after reading about it. When a filter is applied via np.convolve, I wasn't sure that is considered one-pass reverse, one-pass forward, or one-pass forward and reverse.

If this should be one-pass forward, that would mean this filter is also causal? Compared to two-pass (I think that's correct for IIR), which could be non-causal?

Reply:
I think I would just say it's one pass? A one pass filter is not necessarily causal, which is a statement on whether future samples are used in the calculation. What we do is an acausal on-pass, I'm pretty sure (check out a kernel plot - the time definition is positive & negative time around the current sample). 

Also, I think we need to check the "group delay" we report because I think that's the delay if it were a causal filter. In practice we have a zero-phase filter, I believe (cus it's acausal). I might also be getting confused in the terminology here though. 

A two-pass (filt-filt) approach is always acausal, because the reverse pass uses "future" information.

Reply:
I updated this to one-pass. I'm still a bit confused about how to determined direction based on scipy's firwin filter coefficients.

I updated the group delay to zero based on the Cheveigné 2019 paper: "It is delayed if the filter is causal. The delay may be avoided by choosing a zero-phase filter (green), but the response is then acausal."